app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,103 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+
+    if @task.completed
+      @status = "completed"
+    elsif @task.due_date && @task.due_date < DateTime.now
+      @status = "overdue"
+    else
+      @status = "in-progress"
+    end
+
+    if @task[:completed]
+      @change = "Incomplete"
+    else
+      @change = "Complete"
+    end
```

Comment:
You called this out in your comprehension question answers, but a lot of this logic is based around setting up variables for the view, and the view alone. Particularly, setting `@status` and `@change`, I may expect this logic to go into a view helper, which we'll get into depth soon. (The fact that your `index` view has similar logic repeated is a good sign of needing a refactor.) For now, putting it in the controller is fine, but I'm glad that this feels like a code smell for you!